### PR TITLE
Cumulus 1375

### PR DESCRIPTION
**Pull request by laurenfrederick**
**Summary:** Migrate to new Elasticsearch client

Addresses [CUMULUS-1375: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1375)

## Changes

* Updates to migrate away from deprecated ES client to new one. Migration guide: https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/breaking-changes.html

## Test Plan

- [x] Unit tests
- [x] Integration tests
- [x] Dashboard manual testing
- [x] New deployment
- [x] Reindex functionality
- [x] Index from database

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [x] Integration tests

### CODE REVIEW

File: packages/api/es/search.js [hunk 1]
```diff
@@ -34,24 +34,31 @@ const getCredentials = () =>
...
   requestTimeout: 5000
 });
 
 const esProdConfig = async (host) => {
   if (!aws.config.credentials) await getCredentials();
 
+  let node = 'http://localhost:9200';
+
+  if (process.env.ES_HOST) {
+    node = `http://${process.env.ES_HOST}`;
```
Comment (markdboyd) [inline packages/api/es/search.js]: For our lambdas, won't the ES host address be on `https`?

File: packages/api/es/search.js [hunk 2]
```diff
@@ -34,24 +34,31 @@ const getCredentials = () =>
  * @returns {string} elasticsearch local address
  */
 const getLocalEsHost = () => {
-  if (process.env.LOCAL_ES_HOST) return `${process.env.LOCAL_ES_HOST}:9200`;
-  if (process.env.LOCALSTACK_HOST) return `${process.env.LOCALSTACK_HOST}:4571`;
+  if (process.env.LOCAL_ES_HOST) return `http://${process.env.LOCAL_ES_HOST}:9200`;
+  if (process.env.LOCALSTACK_HOST) return `http://${process.env.LOCALSTACK_HOST}:4571`;
```
[the remark this answers is not in the document]
Comment (laurenfrederick) [reply packages/api/es/search.js]: No because `esTestConfig()` below sets up the config object

File: packages/api/es/search.js (hunk 1 above)
Comment (laurenfrederick) [reply packages/api/es/search.js]: Hmmm good question. I did a ton of testing on this and `http` seemed to work fine. I'll try with `https` just to see.

File: packages/api/es/search.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (laurenfrederick) [reply packages/api/es/search.js]: Sorry, I misunderstood your question. I fixed it, thanks!

File: packages/api/es/search.js (hunk 1 above)
Comment (laurenfrederick) [reply packages/api/es/search.js]: I changed it to `https`, but it seems to work either way due to the security group allowing all traffic within the VPC. Both ES and lambda are behind the VPC.

**Pull request merged**